### DISCUSSION TITLE
タグの名寄せリクエスト

### DISCUSSION THREAD

**Discussion by shiracamus** [increments/qiita-discussions#80] [要望・アイデア・不具合]
いくつかのタグが名寄せされていますが、タグ名寄せリストはどこかで見られますでしょうか？
名寄せして欲しいタグがあるので、リクエストしておきます。

C言語 ⇒ C
golang,Go ⇒ Go

**Comment by kskwtnk** [comment]
@shiracamus 

ご意見ありがとうございます！
現状、ユーザーの皆さま向けに公開しているものはありません 🙇 

今回いただいた分は対応いたしますね！

**Comment by kskwtnk** [comment]
@shiracamus こちら、完了いたしました！

**Comment by shiracamus** [comment]
追加リクエストします。

nextjs ⇒ next.js
nuxtjs ⇒ nuxt.js

**Reply by kyntk** [reply]
@shiracamus ありがとうございます! 完了しました!

**Comment by shiracamus** [comment]
カタカナと英語の名寄せもいろいろありそうですね・・・

どれかにまとめたい： データサイエンス, data-science, dataScience 
どっちかにまとめたい： common-lisp, CommonLisp

**Reply by kyntk** [reply]
@shiracamus ありがとうございます!
記事数等加味しながら名寄せします!

**Reply by kyntk** [reply]
データサイエンスは名寄せ完了しました

**Comment by shiracamus** [comment]
あいまい検索で自動抽出して目視確認対応するようなことを考えた方がいいかも・・・

raspberry-pi ⇒ RaspberryPi
raspbery-pi ⇒ RaspberryPi
raspberry ⇒ RaspberryPi  （フォロワーの多い方へ）

**Reply by kyntk** [reply]
@shiracamus こちらもありがとうございます。
自動抽出については、現状すぐに対応するということは考えておりません。
Qiitaとしてはユーザーにより自由にタグを生成できるようにしているので、一定ブレが発生してしまうのは仕方ないと考えております。

運営としてもタイポなどで紛らわしいものを見つけ次第名寄せをしていますが、このようにご報告いただけるのはとてもありがたいです!

**Reply by kyntk** [reply]
こちらも完了しました!

**Comment by munierujp** [comment]
FYI: こんな記事もあります。

[Qiitaの◯◯.jsタグの表記ゆれと最も使われているタグ一覧 - Qiita](https://qiita.com/potato4d/items/23023664dbfdbbd77b98)

JSフレームワークの場合、時期によって正式名称に `.js` がついてたりついてなかったりするのでややこしいですね

**Reply by kyntk** [reply]
@munierujp ありがとうございます!
そうですね、正直名寄せ先にも迷うことが多々あります。
複数のサイトを見てもぶれているものがあったりと、安易に名寄せできないというものもあるのが正直なところです。

**Reply by shiracamus** [reply]
参考までに、質問サイトの teratail.com では、タグを申請する運用になってますね。
回答が付きやすいように回答者がフォローしてるタグに質問を集める、という意図があるのかと思います。

**Reply by kyntk** [reply]
@shiracamus
情報ありがとうございます!

そのほうがタグの管理の面では良さそうですし、探すユーザー側にもメリットが多そうですが、現時点でその変更は難しいなと考えております。

**Comment by kyntk** [comment]
https://github.com/increments/qiita-discussions/discussions/80#discussioncomment-1840081

こちらでいただいていた以下の名寄せですが、記事数が多いのは`common-lisp`で正しくは`Common Lisp`のようで、名寄せ先に迷っており、一旦名寄せをせずにおります。
もし詳しい方がいましたらご意見いただけると嬉しいです!

> どっちかにまとめたい： common-lisp, CommonLisp

**Reply by shiracamus** [reply]
名寄せすると、一つのタグでどちらの記事もまとめて表示される、ということにならなのでしょうか？
まとめて表示されるならどちらでもいい気がしますが、プログラミング言語は大文字で始めてるタグが多いので CommonLisp がよさそうですね。

**Reply by kyntk** [reply]
> 名寄せすると、一つのタグでどちらの記事もまとめて表示される、ということにならなのでしょうか？

1つのタグでまとめて表示されるので、まとめたほうが見つけやすくなります。
通常であれば、正式に近いタグのほうが投稿数が多く、記事数が多いタグに名寄せをすればいいのですが、今回は`common-lisp`のほうが多いので少し迷っているという状態でした!

**Comment by shiracamus** [comment]
名寄せした方が良さそうなタグをリクエストしておきます。
rust と #rust
SplingBoot と spling-boot
ネットワーク と network
メール と mail

相変わらず既存タグがあるのに # や カンマ 付のタグを指定する人がいるので、やはり、新規投稿時のタグ指定の 先頭の# と 最後のカンマ を削除する処理を入れた方がいいと思います。（既存記事には手をいれない）
フォロワーの多い既存タグにまとめた方がタイムラインに表示されて記事が目に止まりやすくなりますので。せっかくの記事がもったいないです。

**Reply by kskwtnk** [reply]
@shiracamus 
ご意見ありがとうございます！

仕組みとして改善したいのと、名付けの変更ルールをどうすべきか……というのがあり、議論いたします 🙏 
キャメルケースとケバブケース、日本語と英語など、モノとしては同じでも表現としては大きめの派閥争いがある分野ですし……
同じタグになっている方が良いのは間違いないのですが、今までルールが無かった分丁寧めなアナウンスなどを実施した方が良いかもしれない、と考えている最中です 🙏

**Comment by shiracamus** [comment]
質問投稿についてもタグの名寄せをしていただけないでしょうか？
タグに #python を指定してある質問がタグフィードに表示されず、気付くことができませんでした。

**Reply by kskwtnk** [reply]
@shiracamus 
お返事が遅くなってしまい申し訳ありません！
名寄せの処理を記事のみに行っていたため、質問にも適用します！

**Reply by wataru86** [reply]
@shiracamus
大変お待たせしました🙇
質問のタグの名寄せの適用が完了しました！

**Comment by vavenger** [comment]
カタカナと英語の名寄せ希望です。

セキュリティ と Security

**Reply by tanaka-asami415** [reply]
@vavenger 
ありがとうございます！🙏
こちらも進めていきますね💪

**Reply by tanaka-asami415** [reply]
@vavenger 
お待たせいたしました！
「Security」へ統一いたしました🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 
すいません。ちょっと質問です。上記についてですが例えばSecurityの場合、これまでセキュリティでタグ付けされていた記事はユーザーランキングから外されますか？それともどこかのタイミングで集計されて正しく計算された上でランキングされますか？
今見ている感じですと、これまでセキュリティでタグ付けされていた記事がランキングに反映されていなように見えたたためです。多分、他のタグも同じですかね？

**Reply by tanaka-asami415** [reply]
@Brutus03
お問い合わせありがとうございます。
こちら調査して、改めてご返信させていただきます🙇💦

**Reply by vavenger** [reply]
[セキュリティー]というタグも使われています。こちらも[Security]への名寄せをご検討ください。

**Reply by tanaka-asami415** [reply]
@vavenger 
ありがとうございます🙏こちらも対応させていただきます！
完了後は、新しいdiscussions #237 でお知らせいたします🙇

**Comment by Brutus03** [comment]
@tanaka-asami415 
ご連絡ありがとうございます。是非、調査及びご対応のほどお願いします。
私はこれまでセキュリティでタグ付けして投稿していましたが本日、タグの見え方が変わったことに気づき、このdiscussionsにたどり着きました。

タグ付けし直した場合は、再集計してランキングにも反映させるのがあるべき姿だと思います。
また、キーワード検索時にタグを入力して、LGYM数順でソートする際にも影響するかと思います。

ご確認お願いします。

**Reply by tanaka-asami415** [reply]
@Brutus03
ご迷惑をおかけし、大変申し訳ありません🙇
調査したところ、正しく集計ができておりませんでした。

正しく集計されるよう修正を進めますので、反映までもうしばらくお待ちいただけると幸いです。
修正が完了しましたら、改めてお知らせさせていただきます。本当に申し訳ございません🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 
ご確認ありがとうございます。上記についてはご連絡お待ちしております。

また、以下建設的な意見として解釈いただけると幸いです。

例えば、今回のタグ寄せについて[Qiita Blog](https://blog.qiita.com/replacing-tag-rule/)に記載がありますが、これを見ただけでは方針が決まったことだけしか分からず、いつから適用されているのかユーザーには分かりません。また、今の運用だとユーザーに対して周知が届いてないので効果が薄いと思います。

従って、今回のタグ寄せに関する明確なアナウンス及び、ガイドラインにも反映するなど、もう少し伝えるための行動や施策が必要ではないかと思いました。ご検討のほどお願いします。

**Reply by Brutus03** [reply]
上記について余談ですが今回のようなユーザーに対する影響が大きい改修や、新機能のリリース情報等についてはブログだけではなく、タイムライン形式でいつリリースされたかが分かるように、リリース情報的なinfoページがフッタにあると良いと思います。

**Reply by tanaka-asami415** [reply]
@Brutus03 
いつも建設的なご意見をいただき、ありがとうございます🙇

名寄せ処理は機械的に一括でしているのではなく、都度少しずつ修正対応している状況ということもあり
ブログでは方針の発表のみに留まっておりました🙇💦

>新機能のリリース情報等についてはブログだけではなく、タイムライン形式でいつリリースされたかが分かるように、リリース情報的なinfoページがフッタにあると良いと思います。

新機能のリリース情報については、Qiita HPヘッダーにある[リリースノート](https://qiita.com/release-notes)から
ご確認いただけるのですが、タグの名寄せはリリースを伴っていないため、ブログでの告知となっておりました🙇

> いつから適用されているのかユーザーには分かりません。

ご指摘の通りユーザーさまに状況がわかりにくくなっていたので
適用状況の確認とリクエストができるDiscussionsを運営で作成いたしました。
[タグの名寄せリクエスト一覧](https://github.com/increments/qiita-discussions/discussions/237)

今後はこちらで、適用状況などをご確認いただけますと幸いです。
ブログ記事への追記はもちろん、Discussionsへの案内を積極的にしていきます🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 

> 新機能のリリース情報については、Qiita HPヘッダーにある[リリースノート](https://qiita.com/release-notes)から
> ご確認いただけるのですが、タグの名寄せはリリースを伴っていないため、ブログでの告知となっておりました🙇

リリースノートというものがあったんですね。こちら存在を知りませんでした。

であれば前のリプライで間違えてフッタと書いてしまいましたが、正しくは画面上部のヘッダにリリースノートを配置し、新着情報がある場合は表示を変えるなどしたら気づきやすいと思います。

> 今後はこちらで、適用状況などをご確認いただけますと幸いです。
> ブログ記事への追記はもちろん、Discussionsへの案内を積極的にしていきます🙇

承知しました。ご対応ありがとうございます。引き続き、どうぞよろしくお願いします。

**Reply by tanaka-asami415** [reply]
@Brutus03 
ご確認いただきありがとうございます🙏

> 正しくは画面上部のヘッダにリリースノートを配置し、新着情報がある場合は表示を変えるなどしたら気づきやすいと思います。

大変失礼いたしました。私もフッターと記載するところを、ヘッダーと書いておりました🙇💦
今回いただいたご意見により、現運用では「折角のリリースが気づかれていない可能性もある」ということがわかりました💦

配置する場所はお約束できないのですが、新しいリリースがあった際には、通知欄等で通知する等
気づいてもらいやすいよう対応していきたいと思います💪貴重なご意見をいつもありがとうございます🙏

Contributionの再計算につきましては、お時間をいただいており申し訳ありません。
修正が完了しましたら、改めてこちらでお知らせをさせていただきます🙇

タグの名寄せについては、新たにDiscussionを作成しましたので、一旦こちらはクローズとさせていただきますが
引き続きコメントはしていただけますので、何かございましたらご連絡ください🙇

**Reply by tanaka-asami415** [reply]
@Brutus03
お待たせして大変申し訳ございません🙇🙇🙇
Contributionの再計算について修正が完了いたしましたので、ご確認いただけますと幸いです🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 
ご連絡ありがとうございます。
すいません。Contributionの再計算した結果が正しいか確認するため、Contributionの計算ロジックを教えてもらうことはできますか？
エビデンスがなく記憶になり恐縮ですが、セキュリティからSecurityへのタグ寄せ前と比較した場合、タグ寄せ前と比べてContributionsの値が半分以上下回っていると思います。（Securityタグのユーザーランキング「全て」を指します）

私には再計算した結果が正しいのか、元の計算結果が誤っていたのか、タグ寄せされたことにより計算ロジックが変わったのか分からないため、確認が難しいです。

つきましては、Contributionの計算ロジックを教えていただけるか、再度ご確認いただけると幸いです。

**Reply by tanaka-asami415** [reply]
@Brutus03 
お手数をおかけしてしまい、大変申し訳ございません🙇💦
確認したところ、キャッシュが残ってしまっており、修正前のランキングが表示されておりました🙇🙇🙇
現在、修正後のランキングを反映させるのを進めておりますので、もうしばらくお待ちいただけると幸いです🙇🙇

**Reply by tanaka-asami415** [reply]
@Brutus03 
キャッシュのクリアが完了いたしました🙇
また、「セキュリティ」タグを名寄せする前のBrutus03 さんのユーザーランキングも確認してみたのですが、
名寄せ後とほぼ同じContributionで、大きな差はないようです🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 
ご対応ありがとうございます。

> 名寄せ後とほぼ同じContributionで、大きな差はないようです🙇

恐縮ですが、正直確認のしようがないので計算ロジックを教えていただくことはできますか？
もし、難しい場合は大丈夫ですが、元々の計算が間違っていたのか、再計算でロジックが変わったかは知りたいです。
言えるのはユーザーランキングの顔ぶれも変化しているので、タグ寄せ後、数字に影響が出ているのは明らかだと思います。

あと、気にしているのは元々、タグ寄せ前のタグで書いていたユーザーがタグ寄せによって、私のように大幅に数字が下がる場合は報われないと思います。他のユーザーからも問い合わせがくる可能性があるため、きちんとケアした方が良いと思います。

**Reply by tanaka-asami415** [reply]
@Brutus03 
こちらこそ、キャッシュが残った状態で修正完了のご連絡をしてしまい、大変失礼いたしました。
現在調査しておりますので、先に計算ロジックについて返答させていただきます🙇

>元々の計算が間違っていたのか、再計算でロジックが変わったか

計算ロジックは、名寄せ前と再計算後での変更はありません。
以下ブログの項目ごとのContribution値の合算になり、
それぞれの項目の数に変化がなければ、Contributionにも変化はありません。
https://help.qiita.com/ja/articles/qiita-contribution

その他につきましては、改めてご連絡させていただきます🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 
ご回答ありがとうございます。
上記、教えていただいたContributionの計算方法より今の数字が正しく、タグ寄せ前の数字が間違ってたと思います。

恐らくタグ寄せ前はユーザーのContributions数をそのまま出力して、ランキング付けされていたのではないでしょうか。
それであれば、Contributions数が大幅に下がったことなどに説明がつきます。

> その他につきましては、改めてご連絡させていただきます🙇

承知しました。

**Reply by tanaka-asami415** [reply]
@Brutus03 
お待たせして大変申し訳ございません。
慎重に調査いたしましたが、タグの名寄せ前で計算が間違っていた可能性はありませんでした。

ランキングのユーザーが変化している原因としては、名寄せ後は「セキュリティ」タグと「Security」タグのcontributionが合算されてのユーザーランキングになり、名寄せ前にはランキングに入っていなかったユーザーがランキングに入っていることが考えられます。

>他のユーザーからも問い合わせがくる可能性があるため、きちんとケアした方が良いと思います。

再計算が正しく行えていなかった不具合などの案内を、 #237  へ追記いたしました。Twitterでもアナウンスいたします。

この度は、ご迷惑をおかけしてしまい大変申し訳ありませんでした🙇

**Reply by Brutus03** [reply]
@tanaka-asami415 

> 慎重に調査いたしましたが、タグの名寄せ前で計算が間違っていた可能性はありませんでした。

>ランキングのユーザーが変化している原因としては、名寄せ後は「セキュリティ」タグと「Security」タグのcontributionが合算されてのユーザーランキングになり、名寄せ前にはランキングに入っていなかったユーザーがランキングに入っていることが考えられます。

エビデンスがない以上、証明のしようがないため、本件についてはクローズで大丈夫です。ご対応ありがとうございました。
最後に恐縮ですが本件のようなユーザーに影響がある変更については、機能リリースでなくても事前に周知した方が、サービスの運用として今より良くなると思います。

以上です。

**Reply by tanaka-asami415** [reply]
@Brutus03 
こちらこそ色々と不手際があり、度々お手数もおかけしてしまい大変申し訳ございませんでした。
スッキリと納得していただけるエビデンスもお出しすることができず、本当に申し訳ないです。

> 機能リリースでなくても事前に周知した方が、

はい。こちらに関しては、色々とご提案いただいていることと合わせて、今後のサービス改善で検討していきます🙇
すぐに改善がお示しできておらず申し訳ありませんが、今後ともQiitaをご愛顧いただけますと幸いです。

**Comment by tanaka-asami415** [comment]
たくさんのリクエストをいただき、ありがとうございます 🙏

タグの名寄せについて状況を確認していただきやすいよう、こちらへ新たにまとめました！
今後はこちらよりリクエストをお送りください！本件はクローズさせていただきます🙇
- [タグの名寄せリクエスト一覧 #237](https://github.com/increments/qiita-discussions/discussions/237)
